Firebase/Auth/Source/MultiFactor/FIRMultiFactor.m

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright 2019 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import "FIRMultiFactor.h"
+#import "FIRMultiFactor+Internal.h"
+
+#import "FIRAuth_Internal.h"
+#import "FIRAuthBackend+MultiFactor.h"
+#import "FIRAuthDataResult_Internal.h"
+#import "FIRMultiFactorInfo+Internal.h"
+#import "FIRStartMfaEnrollmentRequest.h"
+#import "FIRUser_Internal.h"
+#import "FIRMultiFactorSession+Internal.h"
+
+#if TARGET_OS_IOS
+#import "FIRPhoneAuthCredential_Internal.h"
+#import "FIRPhoneMultiFactorAssertion.h"
+#import "FIRPhoneMultiFactorAssertion+Internal.h"
+#endif
+
+NS_ASSUME_NONNULL_BEGIN
+
+static NSString *kEnrolledFactorsCodingKey = @"enrolledFactors";
+
+static NSString *kUserCodingKey = @"user";
+
+@implementation FIRMultiFactor
+
+- (void)getSessionWithCompletion:(nullable FIRMultiFactorSessionCallback)completion {
+  FIRMultiFactorSession *session = [FIRMultiFactorSession sessionForCurrentUser];
+  if (completion) {
+    completion(session, nil);
+  }
+}
+
+- (void)enrollWithAssertion:(FIRMultiFactorAssertion*)assertion
+                displayName:(nullable NSString *)displayName
+                 completion:(nullable FIRAuthVoidErrorCallback)completion {
+#if TARGET_OS_IOS
+  FIRPhoneMultiFactorAssertion *phoneAssertion = (FIRPhoneMultiFactorAssertion *)assertion;
+  FIRAuthProtoFinalizeMfaPhoneRequestInfo *finalizeMfaPhoneRequestInfo =
+      [[FIRAuthProtoFinalizeMfaPhoneRequestInfo alloc] initWithSessionInfo:phoneAssertion.authCredential.verificationID
+                                                          verificationCode:phoneAssertion.authCredential.verificationCode
+                                                               phoneNumber:@""];
```

Comment:
What's the reason for the phone number being empty here?

Reply:
This field is not useful anymore due to backend design change. Removed this property. Thanks!